src/_core/usdPrices.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import invariant from 'tiny-invariant';
+
+import { usePopupSelector } from '../popup/store/react';
+import Background from '../ui/services/Background';
+import { startPolling } from './polling';
+import { useDebouncedValue } from './useDebouncedValue';
+
+const UsdPricesContext = createContext<
+  null | ((assetIds: string[]) => undefined | (() => void))
+>(null);
+
+export function UsdPricesProvider({ children }: { children: ReactNode }) {
+  const [observedAssetIds, setObservedAssetIds] = useState<string[][]>([]);
+  const observedAssetIdsDebounced = useDebouncedValue(observedAssetIds, 100);
+
+  useEffect(() => {
+    if (!observedAssetIdsDebounced.length) {
+      return;
+    }
+
+    return startPolling(5000, () =>
+      Background.updateUsdPricesByAssetIds([
+        ...new Set(observedAssetIdsDebounced.flat()),
```

Comment:
I doubt any other items would ever be added to this array, so may be use `Array.from` instead of spreading onto an empty array?